src/arrayMethodJoin.js

```diff
@@ -4,8 +4,37 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    if (this.length === 0) {
+      return '';
+    }
+
+    let resultString = '';
+    let newSeparator = separator;
+
+    if (this[0] === null || this[0] === undefined) {
+      resultString = '';
+    } else {
+      resultString = this[0].toString();
+    }
```

Comment:
No need for two different checks for first element and the rest, you can start your loop with 0 and check first element same way as other elements